eo-maven-plugin/src/main/java/org/eolang/maven/fp/Footprint.java

```diff
@@ -0,0 +1,198 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.fp;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.cactoos.Func;
+import org.cactoos.func.StickyFunc;
+import org.cactoos.text.TextOf;
+
+/**
+ * Footprint is a function that knows how to properly copy content from source
+ * file to target file.
+ * <p>Footprint works with three places:
+ *  1) source file which is used to extract content from it
+ *  2) target file which is used to store content to it
+ *  3) global cache file which is used in order not to extract content from source
+ *     if it's already exists
+ *  </p>
+ *
+ * <p>General statements:
+ *  1) if target older than source - no extraction is happened
+ *  2) if target younger than source or does not exist - it will be created and filled up.
+ *     It can be created from source, or from global cache if it exists and cacheable and
+ *     older than source.
+ *  3) the cache is updated if it's cacheable AND (it does not exist or if it's younger than source)
+ *  </p>
+ *
+ * <p>Excluding any type of errors there are 4 possible scenarios of Footprint work:
+ *  1) do nothing and just return target file.
+ *  2) extract content, update target and return target. It can be possible if target younger than
+ *  3) extract content, update target, update cache and return target
+ *  4) copy content from cache to target and return target</p>
+ * @since 0.41.0
+ */
+public final class Footprint implements Func<Func<Path, String>, Path> {
+    /**
+     * Absolute path to file with source code.
+     */
+    private final Path source;
+
+    /**
+     * Absolute path to target file.
+     */
+    private final Path target;
+
+    /**
+     * Global cache.
+     */
+    private final Cache cache;
+
+    /**
+     * Ctor.
+     * @param source Absolute source path
+     * @param target Absolute target path
+     * @param cache Cache
+     */
+    public Footprint(
+        final Path source,
+        final Path target,
+        final Cache cache
```

Comment:
@maxonfjvipon the `Cache` class is final, which means that only an instance of that particular class may be passed here. It seems more reasonable to have yet another constructor here, which accepts all three arguments of the `Cache` constructor.

Reply:
@yegor256 then we will have constructor with at least 4 parameters which is prohibited by checkstyle and PMD. This is the only reason we don't have it right now

Reply:
@maxonfjvipon if so, make `Cache` an interface